lib/src/main/kotlin/com/swmansion/starknet/data/types/Contracts.kt

```diff
@@ -124,7 +127,7 @@ data class Cairo0ContractDefinition(private val contract: String) {
 }
 
 @Serializable
-data class Cairo1ContractDefinition(private val contract: String) {
+data class CairoContractDefinition(private val contract: String) {
```

Comment:
I think the way you've done it is ok. Let's just make sure this is properly announced as a breaking change.

Reply:
Come to think of it, can't we consider Cairo 2 to be an extension of Cairo 1?
Then we could still call the old dataclass `Cairo1ContractDefinition` and 
```kotlin
typealias Cairo2ContractDefinition = Cairo1ContractDefinition
``` 
And if something changes and Cairo2 contracts stop being supported via `Cairo1ContractDefinition`, we would remove typealias just redefine `Cairo2ContractDefinition`, while not affecting the old dataclass in any way, both now and in case of such change. 

I'm okay with either approach, but we could avoid making it breaking that way, while not creating any confusion by having a version-less `CairoContractDefinition`. What do you think?

Reply:
for now - yes. how will it look like in the future - I have no idea.

I like both ideas, and think they are both good. Both approaches seems to be future-proof. It all depends on how much you do not want to make breaking changes. Personally, I'm quite ok with this one as it is not very compilated to adjust to. But I understand a need to not introduce breaking changes, so it may make sense to do what you've just suggested. I leave the final decision to you.